aim/sdk/adapters/prophet.py

```diff
@@ -0,0 +1,76 @@
+from sys import stdout
+from typing import Any, Dict, Optional, TypeVar
+
+from aim import Run
+from aim.ext.resource import DEFAULT_SYSTEM_TRACKING_INT
+
+
+Prophet = TypeVar("Prophet")
+
+
+class AimLogger:
+    def __init__(
+        self,
+        prophet_model: Prophet,
+        repo: Optional[str] = None,
+        experiment: Optional[str] = None,
+        system_tracking_interval: int = DEFAULT_SYSTEM_TRACKING_INT,
+        log_system_params: bool = True,
+        log_cout=stdout,
+    ):
+        """
+        AimLogger for Prophet models.
+        Prophet doesn't have a callback system and isn't trained iteratively.
+        Thus, only the hyperparameters and user-provided metrics are logged.
+        """
+        self._repo_path = repo
+        self._experiment = experiment
+        self._system_tracking_interval = system_tracking_interval
+        self._log_system_params = log_system_params
+        self._run = None
+        self._run_hash = None
+        self._log_cout = log_cout
+        self.setup(prophet_model.__dict__)
+
+    @property
+    def experiment(self) -> Run:
+        if not self._run:
+            self.setup()
+        return self._run
+
+    def setup(self, hparams: Dict[str, Any]) -> None:
```

Comment:
I wonder when the setup method is called? Should it be done manually?

Reply:
Nope, it's currently called in the constructor! I thought that would be the easiest way to handle things (requiring minimal input from the user), but please let me know if that's incongruent with the general Aim workflow.